docs/pages/repo/docs/core-concepts/caching.mdx

```diff
@@ -83,6 +83,69 @@ turbo run build --force
 
 Note that `--force` disables cache reads but does not disable cache writes. If you want to disable cache writes, use the `--no-cache` flag.
 
+## Handling Node.js versions
+
+To account for Node.js versions, use [the `engines` key in package.json](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#engines). Turborepo will see the changes to your `package.json` and miss cache when the field is updated.
+
+## Handling platforms and other arbitrary hash contributors
+
+For advanced use cases, you may need to account for operating systems (OS), architecture, or other external factors that you want to contribute to your hash. You can do this in four quick steps.
+
+### 1. Write an arbitrary file to disk
+
+First, create a script that accounts for the hash contributors that you are interested in. For example, here is a Node.js script that looks at platform and architecture and writes those details to a file (`turbo-cache-key.json`):
+
+```js
+#!/usr/bin/env node
+
+const { writeFileSync } = require('fs');
+const { join } = require('path');
+
+const { platform, arch } = process;
+const file = join(__dirname, '..', 'turbo-cache-key.json');
+const str = JSON.stringify({ platform, arch });
+console.log(`Generating cache key: ${str}`);
+writeFileSync(file, str);
+```
+
+### 2. Add the file to your .gitignore
+
+You won't want to commit this file to source control since it's dependent on environment. Add it to your `.gitignore`:
+
+```txt
+// .gitignore
+
++ turbo-cache-key.json
+```
+
+### 3. Add the file to `globalDependencies`
```

Comment:
I would make this a little less prescriptive. It may be that this type of advanced use case only applies to one task (e.g. build and not lint). I would advise the user to be judicious about specifying it as an input where it matters)